subsys/settings/src/settings_file.c

```diff
@@ -360,49 +362,63 @@ static int settings_file_save_and_compress(struct settings_file *cf,
 
 }
 
+static int settings_file_save_start(struct settings_store *cs)
+{
+	struct settings_file *cf = (struct settings_file *)cs;
+	int rc;
+
+	fs_file_t_init(&cf->file);
+	rc = fs_open(&cf->file, cf->cf_name, FS_O_CREATE | FS_O_RDWR);
```

Comment:
The code is always appending to the end of file so the `FS_O_APPEND` could probably be used here, with no need for `fs_seek` invocation bellow.

Reply:
I'm tried to add FS_O_APPEND instead of FS_O_CREATE. The fs_open returned error -2. 
I have not tried to know what was the cause. I can try it again an investigate further.

Reply:
> I'm tried to add FS_O_APPEND instead of FS_O_CREATE. The fs_open returned error -2. I have not tried to know what was the cause. I can try it again an investigate further.

FS_O_CREATE allows fs_open to create file if such does not exist, you should use both.

Reply:
Oh yes, you are correct. FS_O_CREATE I will change the implementation to use the APPEND option instead of seek. That will solve the littlefs seek flush issue. 
Thank you for your input

Reply:
This change could be done to the existing implementation as well. Is this the cause of the slow performance?

Reply:
Partially yes. This needs to be solved as well, otherwise the closing of the file(flushing) wont be effective alone.